cluster/gce/trusty/node.yaml

```diff
@@ -162,8 +163,16 @@ start on stopped kube-env
 
 script
 	. /etc/kube-env
-	# If kubelet or kube-proxy is not installed in the image, pull release binaries and put them in /usr/bin.
-	if ! which kubelet > /dev/null || ! which kube-proxy > /dev/null; then
+	# If both kubelet and kube-proxy test args are set, this is a testing cluster.
+	# We pull binaries and put them in /usr/local/bin. If it is not a testing
```

Comment:
I have an idea: we use an env variable for forcing installation (I guess we need to revise config-test.sh and debian's helper.sh). If it is set, we always pull binaries and place them in /usr/local/bin. If it is not set, we try to make symlinks in /usr/local/bin pointing to the pre-installed binaries (if they exist). When starting kubelet and kube-proxy, we always use /usr/local/bin. With this, we put binary preparation in one particular upstart job, and it is transparent to the upstart jobs of running kubelet and kube-proxy.